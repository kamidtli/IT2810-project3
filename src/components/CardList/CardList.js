import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag';
import Grid from '@material-ui/core/Grid';
import Card from '../Card/Card';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    maxWidth: 1000,
    color: theme.palette.secondary.main,
  },
  gridItem: {
    display: 'flex',
    justifyContent: 'center',
    padding: theme.spacing(2),
  },
  button: {
    margin: theme.spacing(1),
  },
  noResults: {
    height: '500px',
    paddingTop: theme.spacing(10),
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
}));

function CardList(props) {
  const classes = useStyles();
  const { data } = props;

  if (data.length > 0) {
    return (
      <div className={classes.root}>
        <Grid container>
          {data.map((movie) => (
            <Grid item xs={12} md={6} lg={4} key={movie._id} className={classes.gridItem}>
              <Card
                title={movie.title}
                shortDescription={movie.plot}
                imgUrl={movie.poster}
                id={movie._id}
              />
            </Grid>
          ))}
        </Grid>
      </div>
    );
  }

  props.onLastPage(true);
  return (
    <div className={classes.root}>
      <div className={classes.noResults}>
        No more results
      </div>
    </div>
  );
}

const mapStateToProps = (state) => ({
  search: state.search,
});

const mapDispatchToProps = (dispatch) => ({
  onLastPage: (val) => dispatch({ type: 'LAST_PAGE', val }),
});

export default connect(mapStateToProps, mapDispatchToProps)(CardList);
