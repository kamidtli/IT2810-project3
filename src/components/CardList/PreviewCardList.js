import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Card from '../Card/Card';
import search from '../../reducers/search';
import SortBar from '../SortBar/sortBar';
import FilterBar from '../FilterBar/filterBar';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    maxWidth: 1000,
  },
  gridItem: {
    display: 'flex',
    justifyContent: 'center',
    padding: theme.spacing(2),
  },

}));

export default function PreviewCardList() {
  const classes = useStyles();

  const SEARCH_QUERY = gql`
  {
    findMoviesBasedOnYearRange(min:2015, max: 2019, sort:"-released", pagination:6,){
        _id
        title
        plot
        poster
    }
  }
  `;


  const {
    data, loading, error, fetchMore,
  } = useQuery(SEARCH_QUERY);

  if (loading) return <p>LOADING</p>;
  if (error) return <p>{error.message}</p>;


  const renderData = (toRender) => (
    toRender.map((movie) => (
      <Grid item xs={12} md={6} lg={4} key={movie._id} className={classes.gridItem}>
        <Card
          title={movie.title}
          shortDescription={movie.plot}
          imgUrl={movie.poster}
          id={movie._id}
        />
      </Grid>
    ))
  );


  return (
    <div>
      <div className={classes.root}>
        <Grid container>
          {renderData(data.findMoviesBasedOnYearRange)}
        </Grid>
      </div>
    </div>
  );
}
