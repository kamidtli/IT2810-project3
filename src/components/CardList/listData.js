const listData = [
  {
    key: 1,
    img: 'https://images.unsplash.com/photo-1569211675084-8b18179e965c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1354&q=80',
    title: 'Test tittel 1',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 2,
    img: 'https://images.unsplash.com/photo-1567416592215-fb3c07cfb7c5?ixlib=rb-1.2.1&auto=format&fit=crop&w=1947&q=80',
    title: 'Tittel 1234',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 3,
    img: 'https://images.unsplash.com/photo-1569211675084-8b18179e965c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1354&q=80',
    title: 'Once upon a time in Hollywood',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 4,
    img: 'https://images.unsplash.com/photo-1567416592215-fb3c07cfb7c5?ixlib=rb-1.2.1&auto=format&fit=crop&w=1947&q=80',
    title: 'Inception',
    desc: 'Dette er en lengre beskrivelse enn den tidligere korte beskrivelsen. Som tidligere er denne beskrivelsen også av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 5,
    img: 'https://images.unsplash.com/photo-1569211675084-8b18179e965c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1354&q=80',
    title: 'Gemini Man',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 6,
    img: 'https://images.unsplash.com/photo-1569211675084-8b18179e965c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1354&q=80',
    title: 'lettiT',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 7,
    img: 'https://images.unsplash.com/photo-1567416592215-fb3c07cfb7c5?ixlib=rb-1.2.1&auto=format&fit=crop&w=1947&q=80',
    title: 'Joker',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
  {
    key: 8,
    img: 'https://images.unsplash.com/photo-1567416592215-fb3c07cfb7c5?ixlib=rb-1.2.1&auto=format&fit=crop&w=1947&q=80',
    title: 'Zombieland: Double Tap',
    desc: 'Dette er en kort beskrivelse av en god film som finnes på Netflix.com',
    author: 'Kim Midtlid',
    year: 2019,
    genre: 'Action',
    age: 15,
    duration: '1h 57m',
  },
];

export default listData;
