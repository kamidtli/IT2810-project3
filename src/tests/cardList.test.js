import React from 'react';
import { shallow } from 'enzyme';
import { MockedProvider } from '@apollo/react-testing';
import gql from 'graphql-tag';
import CardList from '../components/CardList/CardList';


const SEARCH_QUERY = gql`
    {
        filterMovies (
            searchValue: "pirates}",
            genre: "action",
            yearRange: [1980,2019],
            ratingRange: [5,10],
            sort: "rating", pagination: 12, skip: 0 ) {
            _id
            title
            plot
            poster
            imdb {
                rating
            }
        }
    }
`;

const mocks = [
  {
    request: {
      query: SEARCH_QUERY,
      variables: {
        searchValue: 'pirates',
        genre: '',
        yearRange: [1980, 2019],
        ratingRange: [5, 10],
        sort: 'rating',
        pagination: 12,
        skip: 0,
      },
    },
    result: {
      data: {
        filterMovies: [
          {
            _id: '573a13aaf29313caabd2128b',
            title: 'Pirates of the Caribbean: The Curse of the Black Pearl',
            plot: "Blacksmith Will Turner teams up with eccentric pirate \"Captain\" Jack Sparrow to save his love, the governor's daughter, from Jack's former pirate allies, who are now undead.",
            poster: 'https://m.media-amazon.com/images/M/MV5BNGYyZGM5MGMtYTY2Ni00M2Y1LWIzNjQtYWUzM2VlNGVhMDNhXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX677_AL_.jpg',
            imdb: {
              rating: 8.1,
            },
          },
          {
            _id: '573a13aef29313caabd2ec0c',
            title: "Pirates of the Caribbean: Dead Man's Chest",
            plot: "Jack Sparrow races to recover the heart of Davy Jones to avoid enslaving his soul to Jones' service, as other friends and foes seek the heart for their own agenda as well.",
            poster: 'https://m.media-amazon.com/images/M/MV5BMTcwODc1MTMxM15BMl5BanBnXkFtZTYwMDg1NzY3._V1_SY1000_SX677_AL_.jpg',
            imdb: {
              rating: 7.3,
            },
          },
          {
            _id: '573a13b3f29313caabd3de52',
            title: "Pirates of the Caribbean: At World's End",
            plot: 'Captain Barbossa, Will Turner and Elizabeth Swann must sail off the edge of the map, navigate treachery and betrayal, and make their final alliances for one last decisive battle.',
            poster: 'https://m.media-amazon.com/images/M/MV5BMjIyNjkxNzEyMl5BMl5BanBnXkFtZTYwMjc3MDE3._V1_SY1000_SX677_AL_.jpg',
            imdb: {
              rating: 7.1,
            },
          },
          {
            _id: '573a13c3f29313caabd69a70',
            title: 'Pirates of the Caribbean: On Stranger Tides',
            plot: 'Jack Sparrow and Barbossa embark on a quest to find the elusive fountain of youth, only to discover that Blackbeard and his daughter are after it too.',
            poster: 'https://m.media-amazon.com/images/M/MV5BMjE5MjkwODI3Nl5BMl5BanBnXkFtZTcwNjcwMDk4NA@@._V1_SY1000_SX677_AL_.jpg',
            imdb: {
              rating: 6.7,
            },
          },
        ],
      },
    },
  },
];
describe('Card List', () => {
  it('renders without crashing', () => {
    const cardList = shallow(<MockedProvider mocks={mocks}><CardList /></MockedProvider>);
    expect(cardList.toJSON).toMatchSnapshot();
  });
});
