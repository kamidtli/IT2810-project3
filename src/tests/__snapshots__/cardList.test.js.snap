// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List renders without crashing 1`] = `undefined`;
